modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/EdgeNGramTokenFilterFactory.java

```diff
@@ -77,4 +84,20 @@ public TokenStream create(TokenStream tokenStream) {
     public boolean breaksFastVectorHighlighter() {
         return true;
     }
+
+    @Override
+    public TokenFilterFactory getSynonymFilter(boolean lenient) {
+        if (lenient) {
+            return IDENTITY_FILTER;
```

Comment:
I don't think we should silently pass the `IDENTITY_FILTER`, even in `lenient` mode. Putting an `edge_ngram` before `synonyms` is not supported, we should pass the filter as is for BWC and refuse to build for new indices.